modal/queue.py

```diff
@@ -44,6 +64,21 @@ def main():
 
         assert my_queue.get() == "some value"
         assert my_queue.get() == 123
+
+        my_queue.put(0)
+        my_queue.put(1, partition_key="foo")
+        my_queue.put(2, partition_key="bar")
+
+        # Default and "foo" partition are ignored by the get operation.
+        assert my_queue.get(partition_key="bar") == 2
+
+        # Set custom 10s expiration time on "foo" partition.
```

Comment:
I think I still have the question I asked (poorly) on Slack. What happens if you do this:

```
my_queue.put(1, partition="a", partition_ttl=20)
time.sleep(10)
my_queue.put(2, partition="a", partition_ttl=5)
```

I could imagine at least 2 possible states after the third line executes:
1. The `"a"` partition has 5 seconds to live; we've completely reset its TTL
2. The `"a"` partition has 10 seconds to live;  it lives until at least the longest TTL implied by puts during its lifetime.

IMO if (1) is the answer, I sort of think that's an API footgun, since queues are explicitly meant to used across multiple functions, so it can be hard / impossible from the local context where queue is used to tell whether the requested TTL will actually be honored.

I think another unspecified behavior here is what happens if you set a `partition_ttl` when using the default / anonymous partition.